### PR TITLE
Alignmentsieve (blacklist, tn5 shift, minqual, rm mito reads) + prettier qc report

### PR DESCRIPTION
**Pull request by Maarten-vd-Sande**
**What problem is the PR solving / What's new?**
If genomepy manages to download a blacklist, it makes use of it.

**What did change**
WIP

### CODE REVIEW

File: rules/get_genome.smk [hunk 1]
```diff
@@ -38,7 +38,7 @@ checkpoint get_genome:
         active_plugins=$(genomepy config show | grep -Po '(?<=- ).*' | paste -s -d, -)
         trap "genomepy plugin enable {{$active_plugins,}} >> {log} 2>&1; rm -f {params.temp}" EXIT
 
-        genomepy plugin disable {{blacklist,bowtie2,bwa,star,gmap,hisat2,minimap2}} >> {log} 2>&1
+        genomepy plugin disable {{blacklist,bowtie2,bwa,gaps,gmap,hisat2,minimap2}} >> {log} 2>&1
```
Comment (siebrenf) [inline rules/get_genome.smk]: to prevent bugs after a merge conflict: this must become star :)

File: rules/qc.smk [hunk 2]
```diff
@@ -164,20 +265,18 @@ def get_trimming_qc(sample):
 
 def get_alignment_qc(sample):
```
Comment (siebrenf) [inline rules/qc.smk]: nice. that's a lot cleaner

**Comment by Maarten-vd-Sande** [comment]
Lots of changes, biggest change being that we now can make use of alignmentsieve. This means minimum mapping quality, tn5 shifts, mitochrondiral and blacklist removal. Another new "cool" feature is that our QC reports look much better (if I may say so).

ezgif com-video-to-gif(3)

File: rules/configuration.smk [hunk 3]
```diff
@@ -396,10 +402,16 @@ if 'replicate' in samples:
...
         replicate=any_given('replicate'),
+
 if 'condition' in samples:
     wildcard_constraints:
         sample=any_given('sample', 'condition'),
+        assembly=any_given('assembly'),
+        sorting='coordinate|queryname',
+        sorter='samtools|sambamba'
+else:
         condition=any_given('condition'),
```
Comment (siebrenf) [inline rules/configuration.smk]: What's happening here?

Firstly, the wildcard constraints are compounding, so assembly was already set. 
Second, your adding condition only if it is NOT in samples?
Third, sorting and sorter are only set of condition in samples? 

I think you want this:

```
if 'condition' in samples:
    wildcard_constraints:
        sample=any_given('sample', 'condition'),
        condition=any_given('condition'),

if 'replicate' in samples and 'condition' in samples:
    wildcard_constraints:
        sample=any_given('sample', 'replicate', 'condition'),

if get_workflow() in ['alignment', 'chip_seq', 'rna_seq', 'ataq_seq']:
    wildcard_constraints:
        sorting='coordinate|queryname',
        sorter='samtools|sambamba',
```

sorter and sorting may even be placed in the default constraints i guess

Comment (Maarten-vd-Sande) [reply rules/configuration.smk]: Not sure, I haven't looked at it thoroughly. I guess something in the merge did sth weird. I'll change it.

File: rules/alignment.smk [hunk 4]
```diff
@@ -282,48 +277,144 @@ elif config['aligner'] == 'star' or config.get('quantifier', '') == 'star':
...
+        expand("{log_dir}/alignmentsieve/{{assembly}}-{{sample}}.log", **config)
+    benchmark:
+        expand("{benchmark_dir}/alignmentsieve/{{assembly}}-{{sample}}.benchmark.txt", **config)[0]
+    params:
+        minqual=f"--minMappingQuality {config['min_mapping_quality']}",
+        atacshift="--ATACshift" if config['tn5_shift'] else '',
+        blacklist=lambda wildcards, input: "" if os.stat(input.blacklist[0]).st_size == 0 else \
+                                          f"--blackListFileName {input.blacklist}"
+    conda:
+        "../envs/alignmentsieve.yaml"
```
Comment (siebrenf) [inline rules/alignment.smk]: "../envs/deeptools.yaml" already exists. saves us one env.

**Comment by siebrenf** [comment]
whoah, you merged blacklist into blacklist, instead of develop into blacklist!

**Comment by Maarten-vd-Sande** [comment]
Erhm what? I just pulled remote blacklist into local blacklist.

**Pull request merged**